Library/DiscUtils.Ntfs/BiosParameterBlock.cs

```diff
@@ -191,9 +193,20 @@ internal int CalcRecordSize(byte rawSize)
             return rawSize * SectorsPerCluster * BytesPerSector;
         }
 
+        internal bool IsValidOemId()
+        {
+            return (!String.IsNullOrEmpty(OemId) && String.Compare(OemId, 0, NTFS_OEM_ID, 0, 8) == 0);
```

Comment:
Could you check that the `OemId` length equals the `NTFS_OEM_ID` length first - and then use `NTFS_OEM_ID.Length` instead of `8`?

Reply:
Makes sense, thanks!